orodha-keycloak/orodha_keycloak/connection.py

```diff
@@ -0,0 +1,129 @@
+"""
+This Module contains the KeycloakConnection class which is a facade
+used to interact with a keycloak server via python-keycloak.
+"""
+import orodha_keycloak.connections.admin
+import orodha_keycloak.connections.client
+
+
+class KeycloakConnection:
+    """
+        Facade class used for connecting to, and interacting with keycloak for the Orodha
+    shopping list app.
+
+    Args:
+        connection_kwargs(dict):
+            server_url(str): The url of the server that our keycloak is hosted at
+            username(str): The username of the Admin server_user.
+            password(str): The password of the Admin server_user.
+            realm_name(str): The name of the keycloak realm that we are attempting to access.
+            client_id(str): The keycloak client_id that we are using for the connection.
+            client_secret_key(str): The secret key of the keycloak client.
+
+    Raises:
+        InvalidConnectionException: If the connection variables given are invalid
+            and do not allow connection.
+    """
+
+    def __init__(self, **connection_kwargs):
+        args = {key: value for key, value in connection_kwargs.items()}
+        # client_connection = create_client_connection(args)
+        admin_connection = orodha_keycloak.connections.admin.create_admin_connection(args)
+        client_connection = orodha_keycloak.connections.client.create_client_connection(args)
+
+        self.admin_connection = admin_connection
+        self.client_connection = client_connection
```

Comment:
@Edison-Stuart We can just assign these immediately, no need for the intermediate variable(s).